Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py

```diff
@@ -0,0 +1,602 @@
+        self.tenant_domain = demisto.params().get('share_point_domain')
+        self.access_token = self.get_access_token()
+        self.headers = {'Authorization': f'Bearer {self.access_token}'}
+
+    def http_call(self, *args, **kwargs):
+        demisto.log(f'Sending: ')
+        res = self._http_request(timeout=5, *args, **kwargs)
+        if 'status_code' in res and res.status_code == 401:
+            self.access_token = self.get_access_token()
+            res = self._http_request(*args, **kwargs)
+        return res
+
+    def return_valid_access_token_if_exist_in_context(self):
+        integration_context = demisto.getIntegrationContext()
+        access_token = integration_context.get('access_token')
+        valid_until = integration_context.get('valid_until')
+        if access_token and valid_until:
+            if epoch_seconds() < valid_until:
+                return access_token
+
+    def return_token_and_save_it_in_context(self, access_token_response):
+        access_token = access_token_response.get('access_token')
+
+        if not access_token:
+            return demisto.error('Access Token returned empty')
+        expires_in = access_token_response.get('expires_in', 3595)
+        time_buffer = 5  # seconds by which to shorten the validity period
+        if expires_in - time_buffer > 0:
+            # err on the side of caution with a slightly shorter access token validity period
+            expires_in = expires_in - time_buffer
+
+        demisto.setIntegrationContext({
+            'access_token': access_token,
+            'valid_until': epoch_seconds() + expires_in
+        })
+        return access_token
+
+    def get_access_token(self):
+        context_access_token = self.return_valid_access_token_if_exist_in_context()
+        if context_access_token:
+            return context_access_token
+
+        body = json.dumps({
+            'app_name': APP_NAME,
+            'registration_id': self.auth_id,
+            'encrypted_token': get_encrypted(self.tenant_id, self.enc_key)
+        })
+        try:
+            access_token_res = self._http_request(method='POST', full_url=self.host, data=body,
+                                                  headers={'Accept': 'application/json'}, url_suffix='')
+        except DemistoException as error:
+            title = 'Error in authentication. Try checking the credentials you entered.'
+            raise DemistoException(title, error)
+        else:
+            return self.return_token_and_save_it_in_context(access_token_res)
+
+    def list_tenant_sites(self):
+        url = 'https://graph.microsoft.com/v1.0/sites'
+        query_string = {"search": "*"}
+        return self.http_call(method='GET', full_url=url, headers=self.headers, url_suffix='', params=query_string)
+
+    def list_drives_in_site(self, site_id=None, limit=None, next_page_url=None):
```

Comment:
this function needs documentation. see `get_encrypted()`(in your code) for example